### PR TITLE
[MFTF] Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup

### LINKED ISSUE #31441: [Issue] [MFTF] Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup
This issue is automatically created based on existing pull request: magento/magento2#31418: [MFTF] Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup

---------
### Description (*)
Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup

### Resolved issues:
1. [x] resolves magento/magento2#31441: [MFTF] Adding StorefrontGuestCheckoutProceedToPaymentStepActionGroup

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by sidolov** [comment]
@magento create issue

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8582](https://jira.corp.magento.com/browse/ENGCOM-8582) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

File: app/code/Magento/Checkout/Test/Mftf/Test/OnePageCheckoutWithAllProductTypesTest.xml [hunk 1]
```diff
@@ -187,8 +187,8 @@
         <fillField selector="{{CheckoutShippingGuestInfoSection.telephone}}" userInput="{{CustomerAddressSimple.telephone}}" stepKey="fillPhone" />
 
         <!-- Click next button to open payment section -->
-        <click selector="{{CheckoutShippingGuestInfoSection.next}}" stepKey="clickNextBtn"/>
-        <waitForPageLoad stepKey="waitForShipmentPageLoad"/>
+        <actionGroup ref="StorefrontGuestCheckoutProceedToPaymentStepActionGroup" stepKey="clickNext"/>
```
Comment (gabrieldagama) [inline app/code/Magento/Checkout/Test/Mftf/Test/OnePageCheckoutWithAllProductTypesTest.xml]: Please apply the suggestion to preserve backward-compatibility
```suggestion
        <actionGroup ref="StorefrontGuestCheckoutProceedToPaymentStepActionGroup" stepKey="clickNextBtn"/>
```

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @engcom-Charlie, please have a look at my comment below, thanks!

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

File: app/code/Magento/Checkout/Test/Mftf/Test/OnePageCheckoutWithAllProductTypesTest.xml (hunk 1 above)
Comment (engcom-Hotel) [reply app/code/Magento/Checkout/Test/Mftf/Test/OnePageCheckoutWithAllProductTypesTest.xml]: Has been applied.

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE

**Pull request merged**